src/comms/IComms.h

```diff
@@ -80,45 +80,45 @@ class IComms : public ISPI
       @param bufLength The length of @p buffer.
       @return The operation success state.
      */
-    virtual OpStatus GPIOWrite(const uint8_t* buffer, const size_t bufLength) { return OpStatus::NotImplemented; };
+    virtual OpStatus GPIOWrite(const uint8_t* buffer, const size_t bufLength) = 0;
```

Comment:
Why? what's the benefit of this?
IComms is abstract class of what operations could be performed, it's not a requirement that all methods need to have an implementation. So by default all of them are reporting that they are not implemented.

The side effect of this change is that now every class that provides only a part of this interface, has to explicitly add overrides for all other methods just to indicate that, resulting in a lot of useless code spam.

Reply:
Original idea was to just a nice clean interface for [C++ Core Guidelines point C.121](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c121-if-a-base-class-is-used-as-an-interface-make-it-a-pure-abstract-class), however, reverting this to not have much spamminess.